pkg/services/ngalert/store/alertmanager.go

```diff
@@ -110,9 +110,24 @@ func (st DBstore) SaveAlertmanagerConfigurationWithCallback(ctx context.Context,
 // UpdateAlertmanagerConfiguration replaces an alertmanager configuration with optimistic locking. It assumes that an existing revision of the configuration exists in the store, and will return an error otherwise.
 func (st *DBstore) UpdateAlertmanagerConfiguration(ctx context.Context, cmd *models.SaveAlertmanagerConfigurationCmd) error {
 	return st.SQLStore.WithTransactionalDbSession(ctx, func(sess *db.Session) error {
+		newConfigHash := fmt.Sprintf("%x", md5.Sum([]byte(cmd.AlertmanagerConfiguration)))
```

Comment:
I think we can use `ConfiguratioHash` here which should be the same as you calculate here. I there a reason you calculate it here again?

Reply:
I just hoisted the place where it was previously calculated up so we could do the comparison before creating the `config` object. It should still only be calculated one time I think.